### ISSUE TITLE
Solar value in order of 1e3 instead of 1e0

### ISSUE THREAD

**Issue by YoungFaithful** [holgerteichgraeber/ClustForOpt.jl#48]
Clustering result (k_means, centroid) for GER_18 makes no sense. (Tested clust-itr 10-50e3)

**Comment by YoungFaithful** [comment]
![plot3](https://user-images.githubusercontent.com/10561245/54962331-9b540280-4f21-11e9-921e-5121b1d76907.png)
Assumption that el_demand and solar value switched

**Comment by YoungFaithful** [comment]
`
using ClustForOpt
using Plots

ts_input_data, = load_timeseries_data("CEP", "GER_18"; K=365, T=24)
clust_data=run_clust(ts_input_data;method="kmeans",n_init=10000,n_clust=5).best_results.data

function get_attribute(data, attribute)
    data_attribute=Dict{String,Any}()
    for (k,v) in data
        if occursin(attribute, k)
            data_attribute[k]=v
        end
    end
    return data_attribute
end

function plot_attribute(data, attribute)
    att_max=Array{Float64,1}()
    att_min=Array{Float64,1}()
    xticks=Array{String,1}()
    data_attribute=get_attribute(data, attribute)
    for k in sort(collect(keys(data_attribute)))
        push!(att_max, maximum(data_attribute[k]))
        push!(att_min, minimum(data_attribute[k]))
        push!(xticks, split(k,"-")[2])
    end
    att=[att_min att_max]
    plot(att, xticks=(1:length(att_max),xticks), xrotation=60, ylabel=attribute, label=["min", "max"])
end

gr(size=(1000,1000))
plots=[]
    for attribute in ["solar", "wind", "el_demand"]
        push!(plots, plot_attribute(ts_input_data.data, attribute))
        push!(plots, plot_attribute(clust_data, attribute))
    end
    ploti=plot(plots...,layout=grid(3,2))

savefig(ploti, normpath(joinpath(dirname(@__FILE__),"plot.svg")))
`

**Comment by YoungFaithful** [comment]
[study_solarclust.jl.txt](https://github.com/holgerteichgraeber/ClustForOpt.jl/files/3005995/study_solarclust.jl.txt)

**Comment by YoungFaithful** [comment]
@holgerteichgraeber found the problem: dictionaries aren't sorted, that's though the assuption in ClustDataMerged & Co. I'm not exactly sure, how to solve that issue best.

**Comment by holgerteichgraeber** [comment]
Awesome. Can you point to which Dicts are the problem, then I'll have a look tomorrow. Is this only happening for different regions, or also for different attributes?

**Comment by YoungFaithful** [comment]
I think it was coincidence that it didn't happen with different attributes, because that weren't so many. I am not exactly sure, which line is the issue, but I'll add some code:
`"""
    ClustDataMerged(data::ClustData)
constructor 2: convert ClustData into merged format
"""
function ClustDataMerged(data::ClustData)
 n_datasets = length(keys(data.data))
 data_merged= zeros(data.T*n_datasets,data.K)
 data_type=String[]
 i=0
 for (k,v) in data.data
   i+=1
   data_merged[(1+data.T*(i-1)):(data.T*i),:] = v
   push!(data_type,k)
 end
 if maximum(data.deltas)!=1
   throw(@error "You cannot recluster data with different Δt")
 end
 ClustDataMerged(data.region,data.years,data.K,data.T,data_merged,data_type,data.weights,data.mean,data.sdv,data.deltas,data.k_ids)
end
`

**Comment by YoungFaithful** [comment]
```julia
"""
    ClustData(data::ClustDataMerged)
constructor 3: Convert ClustDataMerged to ClustData
"""
function ClustData(data::ClustDataMerged)
 data_dict=Dict{String,Array}()
 i=0
 for k in data.data_type
   i+=1
   data_dict[k] = data.data[(1+data.T*(i-1)):(data.T*i),:]
 end
 ClustData(data.region,data.years,data.K,data.T,data_dict,data.weights,data.mean,data.sdv,data.deltas,data.k_ids)
end
```
vielleicht hilft eine Iteration über data_type?

**Comment by holgerteichgraeber** [comment]
Closed with #67

**Comment by YoungFaithful** [comment]
Problem is back in 0.4.0
![image](https://user-images.githubusercontent.com/10561245/60125680-29b66b00-9741-11e9-93f6-1b3973c437ae.png)

**Comment by holgerteichgraeber** [comment]
closed with #90
